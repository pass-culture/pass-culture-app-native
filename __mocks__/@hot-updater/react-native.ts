import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
