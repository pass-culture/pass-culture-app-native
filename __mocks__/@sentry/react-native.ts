import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
