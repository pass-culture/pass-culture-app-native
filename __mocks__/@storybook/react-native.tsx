import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
