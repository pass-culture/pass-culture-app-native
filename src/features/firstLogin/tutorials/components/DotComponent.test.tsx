import { render, waitFor } from '@testing-library/react-native'
import React from 'react'

import {
  DotComponent,
  DotComponentProps,
} from 'features/firstLogin/tutorials/components/DotComponent'

describe('<DotComponent />', () => {
  it('should render correctly', async () => {
    const renderAPI = renderDotComponent()
    await waitFor(() => {
      expect(renderAPI).toMatchSnapshot()
    })
  })
  it('renders dot active color', async () => {
    const renderAPI = renderDotComponent({
      isActive: true,
    })
    await waitFor(() => {
      expect(renderAPI).toMatchSnapshot()
    })
  })
})

function renderDotComponent(props: DotComponentProps = {}) {
  return render(<DotComponent {...props} />)
}
