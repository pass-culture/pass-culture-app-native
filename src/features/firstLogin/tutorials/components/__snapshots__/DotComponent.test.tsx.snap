// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DotComponent /> renders dot active color 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "marginLeft": 8,
        "marginRight": 8,
      },
    ]
  }
>
  <View
    fill="#eb0055"
    height={12}
    testID="dot-icon"
    width={12}
  >
    dot-icon-SVG-Mock
  </View>
</View>
`;

exports[`<DotComponent /> should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "marginLeft": 8,
        "marginRight": 8,
      },
    ]
  }
>
  <View
    fill="#c7c7cc"
    height={8}
    testID="dot-icon"
    width={8}
  >
    dot-icon-SVG-Mock
  </View>
</View>
`;
